addons/spreadsheet_dashboard/static/src/bundle/chart/chart_type_switcher/chart_type_switcher.js

```diff
@@ -0,0 +1,110 @@
+import { Component, useState } from "@odoo/owl";
+import { registries, readonlyAllowedCommands } from "@odoo/o-spreadsheet";
+import { usePopover } from "@web/core/popover/popover_hook";
+
+const { chartSubtypeRegistry, chartRegistry } = registries;
+
+// ADRM TODO: not a fan ...
+readonlyAllowedCommands.add("UPDATE_CHART");
```

Comment:
Not great to allow some random commands in read-only, but I don't think we have a choice ? 😔

Reply:
Not a great fan, but as you said we do not have a lot of choice. IMO we can live with that for now, anyway it will not impact other users/spreadsheet data